api/app/services/privacy_retention_service.py

```
@@ -0,0 +1,1930 @@
+        try:
+            if not self._table_exists(connection, "translations"):
+                report.stores["translation_cache"] = RetentionStoreResult(
+                    0, None, self._window_seconds
+                )
+                return
+            cutoff_epoch = int((now - timedelta(days=self.retention_days)).timestamp())
+            now_epoch = int(now.timestamp())
+            condition = "created_at < ? OR expires_at <= ?"
+            count = int(
+                connection.execute(
+                    f"SELECT COUNT(*) FROM translations WHERE {condition}",
```

Ruff (0.15.21)

[error] Possible SQL injection vector through string-based query construction

(S608)